moler/events/unix/ping_response.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+
+__author__ = 'Agnieszka Bylica, Tomasz Krol'
+__copyright__ = 'Copyright (C) 2019, Nokia'
+__email__ = 'agnieszka.bylica@nokia.com, tomasz.krol@nokia.com'
+
+
+import re
+import datetime
+from moler.events.lineevent import LineEvent
+
+
+class PingResponse(LineEvent):
+    def __init__(self, connection, till_occurs_times=-1):
+        """
+        Event for 'Wait for response from ping.'.
+        :param connection: moler connection to device, terminal when command is executed
+        :param till_occurs_times: number of event occurrence
+        """
+        super(PingResponse, self).__init__(connection=connection, till_occurs_times=till_occurs_times)
+        self.connection = connection
+        self.detect_pattern = r'\d+\s+bytes\s+from.+'
```

Comment:
Please make shared regex among all instances of this class.

Reply:
detect_pattern is used only in on_new_line so why should be an instance variable instead of class variable?
And why do you derived from LineEvent if you provide your own on_new_line? If you want to use LineEvent as base class then do not write your own on_new_line or you can use TextualEvent as base class and then you can write your own on_new_line.

Reply:
Fixed. Thank you for help.